### ISSUE TITLE
Add `eslint-plugin-import`

### ISSUE THREAD

**Issue by voxpelli** [neostandard/neostandard#15]
Either when https://github.com/import-js/eslint-plugin-import/issues/2948 is fixed or switch to https://github.com/un-ts/eslint-plugin-import-x

**Comment by jerome-benoit** [comment]
If the goal is to support both JS and TS code base: 
  - https://github.com/import-js/eslint-plugin-import or https://github.com/un-ts/eslint-plugin-import-x integration
  - https://github.com/import-js/eslint-import-resolver-typescript integration

AFAIK, the three of them still do not support ESLint flat configuration.

**Comment by voxpelli** [comment]
Yeah, should support both JS and TS 👍

**Comment by bcomnes** [comment]
Doesn't tsc provide a similar check these days? (in ts as well as js files).

**Comment by jerome-benoit** [comment]
> Doesn't tsc provide a similar check these days? (in ts as well as js files).

Integrating them in an eslint based tool is more or less redoing an eslint plugin using tsc behind the scene, which will be some supplemental maintenance work without guarantee that this tsc specific usage will scale.

**Comment by bcomnes** [comment]
I don't mean wrap tsc with eslint, I mean when one is using tsc in parallel with eslint, the tsc lsp will mark unresolved imports, which may duplicate eslint-plugin-import's checks.

**Comment by Uzlopak** [comment]
Maybe use one of the eslint-plugin-import forks, which removed the bloatware?

**Comment by voxpelli** [comment]
> Maybe use one of the eslint-plugin-import forks, which removed the bloatware?

Yeah, just need to wait for https://github.com/un-ts/eslint-plugin-import-x/pull/122 to be finished first

---

Could potentially also re-evaluate the import rules and for each one see if there's an alternative plugin, if we could/should skip the rule, or if we could build a plugin of our own:

* [`import/export`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/export.md)
* [`import/first`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/first.md)
* [`import/no-absolute-path`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-absolute-path.md)
* [`import/no-duplicates`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-duplicates.md)
* [`import/no-named-default`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-named-default.md)
* [`import/no-webpack-loader-syntax`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-webpack-loader-syntax.md)

**Comment by jerome-benoit** [comment]
> I don't mean wrap tsc with eslint, I mean when one is using tsc in parallel with eslint, the tsc lsp will mark unresolved imports, which may duplicate eslint-plugin-import's checks.

The import related rules could be disabled with a tunable. But it start to make quite a lot of boolean tunables. 
Using named configuration as tunable can reduce the ones exposed publicly, but the namespace and its content need to be thoroughly consistent: 
- Orthogonality
-

**Comment by kouts** [comment]
Flat config is now supported by eslint-plugin-import-x@4.1.0 https://github.com/un-ts/eslint-plugin-import-x/pull/131

**Comment by tagliala** [comment]
Hello,

eslint-plugin-import 2.31.0 with flat config and eslint 9 support has been released

**Comment by voxpelli** [comment]
If someone would like to contribute a solution that sets up [eslint-import-resolver-typescript](https://github.com/import-js/eslint-import-resolver-typescript) and [eslint-plugin-import-x](https://github.com/un-ts/eslint-plugin-import-x) that would be appreciated 🙏 

I'm a bit on the fence on this as `eslint-plugin-import` historically is one of those plugins that can truly hurt the performance of ESLint, especially on large projects and especially when combined with TypeScript

---

Would also love to hear from people which of the `import/` rules from `standard` that they are feeling that they miss in `neostandard` right now:

* [`import/export`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/export.md)
* [`import/first`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/first.md)
* [`import/no-absolute-path`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-absolute-path.md)
* [`import/no-duplicates`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-duplicates.md)
* [`import/no-named-default`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-named-default.md)
* [`import/no-webpack-loader-syntax`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-webpack-loader-syntax.md)

Eg. https://github.com/standard/standard/issues/1535 mentions that they think [`import/no-absolute-path`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-absolute-path.md) is bad and should _not_ be added.

Are there any reverse opinions on any of the rules? That we are really missing out on something useful?

**Comment by voxpelli** [comment]
### [import/export](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/export.md)

Ensures there are no duplicate named or `default` exports – kind of useful, especially if one does not run TS one ones files (TS will report the same)

### [import/first](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/first.md)

This one enforces all imports to be placed in the top of the file – I really like this one actually

### [import/no-absolute-path](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-absolute-path.md)

See https://github.com/standard/standard/issues/1535 – I have no strong opinion in either direction – it can be useful to catch cases where one meant to write `./foo` but instead accidentally wrote `/foo`, but the cases where that will not be apparent through other means would be slim, and if anyone is actually intentionally adds a full absolute path, then they surely intend to add it in 95% of the cases? Or is there a worry that eg. junior develops could be confused to add this?

### [import/no-duplicates](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-duplicates.md)

Slightly helpful to catch where one has added the same import multiple times. I'm neutral to positive on it.

### [import/no-named-default](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-named-default.md)

This one I actually have no opinion on, it bans doing `import { default as foo}` and enforces `import foo`. Its a nice to have, nothing else

### [import/no-webpack-loader-syntax](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-webpack-loader-syntax.md)

This one I feel is a bit odd in this context – its webpack specific and webpack is not at all as prevalent.

**Comment by voxpelli** [comment]
> I'm a bit on the fence on this as `eslint-plugin-import` historically is one of those plugins that can truly hurt the performance of ESLint, especially on large projects and especially when combined with TypeScript

This is probably mostly when one adds rules like `import/no-unresolved` and such though, as it needs to do additional lookups and can't rely on purely the AST

**Comment by jerome-benoit** [comment]
Since no stable release is here, you can divide into steps: 

- reach the functional equivalence with standard: same plugins (if relevant for eslint 9), same rules
- see how it goes with `eslint-plugin-import` with user feedback on subsequent releases
- optimize with eventual replacement plugins triggering negative feedback

**Comment by dead-claudia** [comment]
There are a few reports of brokenness in the v9 support (example: https://github.com/import-js/eslint-plugin-import/issues/3079), so it might not be possible to re-add *all* of it quite yet.

**Comment by dead-claudia** [comment]
@voxpelli I *believe* those rules in particular should be usable. Can't find anything to suggest otherwise.
